libsolidity/ast/AST.h

```diff
@@ -849,6 +853,7 @@ class FunctionDefinition: public CallableDeclaration, public StructurallyDocumen
 
 private:
 	StateMutability m_stateMutability;
+	bool m_free;
```

Comment:
Just as a random note without necessarily suggesting to do this in this PR, but if we went ahead with a suggestion like https://github.com/ethereum/solidity/pull/9364#discussion_r459380693 , i.e. assign scopes during parsing already, then we wouldn't need this member, but could derive it from the scope instead...